src/wayland/data_device/mod.rs

```diff
@@ -99,6 +103,9 @@ use seat_data::{SeatData, Selection};
 /// Events that are generated by interactions of the clients with the data device
 #[allow(unused_variables)]
 pub trait DataDeviceHandler: Sized + ClientDndGrabHandler + ServerDndGrabHandler {
+    /// UserData attached to server-side selections
+    type SelectionUserData: Clone + Send + Sync + 'static;
```

Comment:
So, in a very "thinking broad" way, would it make sense to not restrict it do server-side selections, but rather allow it for all selections, adding a `Default` bound?

Basically, this increases the realm of possibles for users, and if they only want to attach data to server-side selections they can use an `enum` to distinguish. So that would be a little less practical.

I'm just throwing the idea as it came to me, no idea if that'd actually be worth it or not.

Reply:
Resolving, as we were not able to come up with a use-case.